### PR TITLE
feat: Fix and cover tests for target x86_64-unknown-linux-musl 

### PR DESCRIPTION
**Pull request by Xuanwo**
This PR addresses the following problems:

- Tests and examples not compilable under `x86_64-unknown-linux-musl`
- Add tests in GitHub actions so that all PRs are covered.

### CODE REVIEW

**Comment by Xuanwo** [comment]
Please squash the commits while merging, thanks!

File: src/profiler.rs [hunk 1]
```diff
@@ -183,12 +183,12 @@ fn write_thread_name_fallback(current_thread: libc::pthread_t, name: &mut [libc:
...
 }
 
-#[cfg(not(any(target_os = "linux", target_os = "macos")))]
+#[cfg(not(all(any(target_os = "linux", target_os = "macos"), target_env = "gnu")))]
 fn write_thread_name(current_thread: libc::pthread_t, name: &mut [libc::c_char]) {
     write_thread_name_fallback(current_thread, name);
 }
 
-#[cfg(any(target_os = "linux", target_os = "macos"))]
+#[cfg(all(any(target_os = "linux", target_os = "macos"), target_env = "gnu"))]
```
[the remark this answers is not in the document]
Comment (Xuanwo) [reply src/profiler.rs:191]: This part is a bit tricky.

`cargo build` works, but `cargo test` will have the following error:

```shell
  = note: /usr/bin/ld: /home/xuanwo/Code/tikv/pprof-rs/target/x86_64-unknown-linux-musl/debug/deps/pprof-f3595f784aae57b5.1iodjb8b0079y0t7.rcgu.o: in function `pprof::profiler::write_thread_name':
          /home/xuanwo/Code/tikv/pprof-rs/src/profiler.rs:194: undefined reference to `pthread_getname_np'
          collect2: error: ld returned 1 exit status

  = help: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-link-libkindname)
```

[the remark this answers is not in the document]
Comment (Xuanwo) [reply src/profiler.rs:191]: @YangKeao Hi, can you take a look at this PR again? I'm willing to change the code if there are other ways to address this issue.

[the remark this answers is not in the document]
Comment (YangKeao) [reply src/profiler.rs:191]: It's pretty wired. I run `nm libc.a |grep pthread|grep getname` in the alpine, and got:

```
0000000000000000 T pthread_getname_np
```

But I came across the same issue you faced. I will approve and merge this PR to make it work under musl (and I'll investigate further why it cannot link with this symbol)

**Review by YangKeao — APPROVED** [review]
LGTM

**Comment by Xuanwo** [comment]
How about creating an issue for the unresolved problem?

**Pull request merged**